packages/extension/src/view/devtools/stateProviders/webStories/webStoriesProvider.tsx

```diff
@@ -0,0 +1,517 @@
+    }
+
+    setPageNumber(pageNumber + 1);
+  }, [pageNumber]);
+
+  const iframeLoadedCallback = useCallback(() => {
+    setIframeLoaded(true);
+  }, []);
+
+  useEffect(() => {
+    window.document.addEventListener(
+      'webStoriesLightBoxEvent',
+      webStoriesLightBoxCallback,
+      false
+    );
+
+    window.document.addEventListener(
+      'iframeLoaded',
+      iframeLoadedCallback,
+      false
+    );
+
+    window.document.addEventListener(
+      'loadMoreData',
+      webStoriesLoadMoreData,
+      false
+    );
+
+    return () => {
+      window.document.removeEventListener(
+        'iframeLoaded',
+        iframeLoadedCallback,
+        false
+      );
+
+      window.document.removeEventListener(
+        'webStoriesLightBoxEvent',
+        webStoriesLightBoxCallback,
+        false
+      );
+
+      window.document.removeEventListener(
+        'loadMoreData',
+        webStoriesLoadMoreData,
+        false
+      );
+    };
+  }, [
+    iframeLoadedCallback,
+    webStoriesLightBoxCallback,
+    webStoriesLoadMoreData,
+  ]);
+
+  const toggleFilterSelection = useCallback(
+    (filterKey: string, filterValue: string) => {
+      setSelectedFilters((prevSelectedFilters) => {
+        const filterIndex = prevSelectedFilters.findIndex(
+          (filter) => filter.key === filterKey
+        );
+
+        if (filterIndex === -1) {
+          return [
+            ...prevSelectedFilters,
+            {
+              key: filterKey,
+              title:
+                filters.find((filter) => filter.key === filterKey)?.title || '',
+              values: [filterValue],
+            },
+          ];
+        }
+
+        const filter = prevSelectedFilters[filterIndex];
+        const newFilterValues = filter.values.includes(filterValue)
+          ? filter.values.filter((value) => value !== filterValue)
+          : [...filter.values, filterValue];
+
+        const newFilter = {
+          ...filter,
+          values: newFilterValues,
+        };
+
+        const newSelectedFilters = [...prevSelectedFilters];
+        newSelectedFilters[filterIndex] = newFilter;
+
+        return newSelectedFilters;
+      });
+
+      setSelectedFilterValues((prevSelectedFilterValues) => {
+        const filter = prevSelectedFilterValues[filterKey] || [];
+
+        const newSelectedFilterValues = {
+          ...prevSelectedFilterValues,
+          [filterKey]: filter.includes(filterValue)
+            ? filter.filter((value) => value !== filterValue)
+            : [...filter, filterValue],
+        };
+
+        return newSelectedFilterValues;
+      });
+
+      setPageNumber(1);
+    },
+    [filters]
+  );
+
+  const resetFilters = useCallback(() => {
+    setSelectedFilters([]);
+    setSelectedFilterValues({});
+    setPageNumber(1);
+  }, []);
+
+  const fetchCategories = useCallback(async () => {
+    const response = await fetch(
+      'https://privacysandbox-stories.com/wp-json/web-stories/v1/web_story_category'
```

Comment:
Shouldn't we use a constant for https://privacysandbox-stories.com/wp-json/web-stories/v1 ?

Reply:
Resolved in latest commits